c/src/point_one/polaris/polaris.c

```diff
@@ -1385,7 +1385,7 @@ void P1_PrintData(const uint8_t* buffer, size_t length) {
     }
 
     str_length +=
-        snprintf(str + str_length, sizeof(str - str_length), "%02x", buffer[i]);
+        snprintf(str + str_length, sizeof(str) - str_length, "%02x", buffer[i]);
```

Comment:
this is definitely more correct but also possibly bad if the str_length variable > 64. Naively, if length is 1000, str_length will be 1000/16 which >> 64.